src/main/scala/fi/oph/koski/valpas/opiskeluoikeusrepository/ValpasRajapaivatService.scala

```diff
@@ -21,6 +21,18 @@ abstract class ValpasRajapäivätService(config: Config) extends Logging {
   def oppivelvollisuusAlkaa(syntymäpäivä: LocalDate): LocalDate =
     oppivelvollisuusAlkaaPäivämäärä.withYear(syntymäpäivä.getYear + oppivelvollisuusAlkaaIka)
 
+  def oppivelvollisuusVoimassaAstiIänPerusteella(syntymäpäivä: LocalDate): LocalDate = {
+    // Huom! Sama logiikka on myös SQL:nä Oppivelvollisuustiedot-luokan luomassa materialized view:ssä.
+    // Varmista muutosten jälkeen, että logiikka säilyy samana.
+    syntymäpäivä.plusYears(oppivelvollisuusLoppuuIka.toLong)
+  }
```

Comment:
Tämä muuten ei vastaa nyt sql:ssä olevaa logiikkaa, koska siellä vähennetään rajapäivästä yksi päivä.

Tällaisenaan funktioiden nimet pitäisi olla esimerkiksi `oppivelvollisuudenPäättymispäiväIänPerusteella`.

Näyttäisi myös, että päätelty oppivelvollisuus voimassa asti -päivämäärä on väärin tämän takia kälissä ja varmaankin myös muissa sitä palauttavissa rajapinnoissa. Esimerkiksi oppijasivulla päivämäärä käyttäytyy näin suhteessa hetuun:

Valppaasta löytyvä henkilö 221105A3023 -> "Oppivelvollisuus: 21.11.2023 asti"
Ei-löytyvä henkilö 110405A6951 -> "Oppivelvollisuus:11.4.2023 asti"

Reply:
En nyt muuta noita metodinimiä, koska tätä "asti"-termiä käytetään käyttöliittymässäkin asti ja kaikissa tietorakenteissa.